src/sympc/protocol/aby3/aby3.py

```diff
@@ -204,3 +206,86 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
         return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def local_decomposition(
+        x: ReplicatedSharedTensor, ring_size: str
+    ) -> List[ReplicatedSharedTensor]:
+        """Performs local decomposition to generate shares of shares.
+
+        Args:
+            x (ReplicatedSharedTensor) : input RSTensor.
+            ring_size (str) : Ring size to generate decomposed shares in.
+
+        Returns:
+            List[ReplicatedSharedTensor]: Decomposed shares in the given ring size.
+
+        Raises:
+            ValueError: If RSTensor does not have session uuid.
+        """
+        if x.session_uuid is None:
+            raise ValueError("Input RSTensor should have session_uuid")
+
+        session = get_session(x.session_uuid)
+        ring_size = int(ring_size)
+        tensor_type = get_type_from_ring(ring_size)
+        rank = session.rank
+        nr_parties = session.nr_parties
+
+        zero = torch.zeros(x.shares[0].shape).type(tensor_type)
+
+        shares = [[zero.clone(), zero.clone()] for i in range(nr_parties)]
+
+        shares[rank][0] = x.shares[0].clone().type(tensor_type)
+
+        shares[(rank + 1) % nr_parties][1] = x.shares[1].clone().type(tensor_type)
+
+        rst_list = []
+        for i in range(nr_parties):
+            rst = x.clone()
+            rst.shares = shares[i]
+            rst.ring_size = ring_size
+            rst_list.append(rst)
+
+        return rst_list
+
+    @staticmethod
+    def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
+        """Perfom ABY3 bit injection for conversion of binary share to arithmetic share.
+
+        Args:
+            x (MPCTensor) : MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+            ring_size (int) : Ring size of arithmetic share to convert.
+
+        Returns:
+            arith_share (MPCTensor): Arithmetic shares of bit in input ring size.
+
+        Raises:
+            ValueError: If input tensor is not binary shared.
+        """
+        input_ring = int(x.share_ptrs[0].get_ring_size().get_copy())  # input ring_size
+        if input_ring != 2:
+            raise ValueError("Bit injection works only for binary rings")
+
+        args = [[share, str(ring_size)] for share in x.share_ptrs]
+
+        decompose = parallel_execution(ABY3.local_decomposition, session.parties)(args)
+
+        # Using zip for grouping on pointers is compute intensive.
+        x1_sh = []
+        x2_sh = []
+        x3_sh = []
+
+        for sh in decompose:
+            x1_sh.append(sh[0].resolve_pointer_type())
+            x2_sh.append(sh[1].resolve_pointer_type())
+            x3_sh.append(sh[2].resolve_pointer_type())
+
+        x1 = MPCTensor(shares=x1_sh, session=session, shape=x.shape)
+        x2 = MPCTensor(shares=x2_sh, session=session, shape=x.shape)
+        x3 = MPCTensor(shares=x3_sh, session=session, shape=x.shape)
+
+        arith_share = x1 ^ x2 ^ x3
```

Comment:
Q: Here you can also use "+", right? (since they are in ring size 2 it should not matter)

Reply:
No george ,here it is session ring size or prime ring size,

Reply:
`+` will not work george